packages/subscription-manager/src/components/SubscriptionManagerContainer/SubscriptionManagerContainer.tsx

```diff
@@ -12,18 +12,57 @@ export type SubscriptionManagerContainerProps = {
 	children?: React.ReactNode;
 };
 
+const getEmailAddress = () => {
+	const subkey: string | undefined = document.cookie
+		?.split( ';' )
+		?.map( ( c ) => c.trim() )
+		?.find( ( c ) => c.startsWith( 'subkey=' ) )
+		?.split( '=' )[ 1 ];
+
+	if ( ! subkey ) {
+		return null;
+	}
+
+	const decodedSubkeyValue = decodeURIComponent( subkey );
+
+	const firstPeriodIndex = decodedSubkeyValue.indexOf( '.' );
+	if ( firstPeriodIndex === -1 ) {
+		return null;
+	}
+
+	const emailAddress = decodedSubkeyValue.slice( firstPeriodIndex + 1 );
+	return emailAddress;
+};
+
 const SubscriptionManagerContainer = ( { children }: SubscriptionManagerContainerProps ) => {
 	const translate = useTranslate();
 
+	const getSubHeaderText = () => {
```

Comment:
Would you turn the `getSubHeaderText` into a callback?

Reply:
I might move this out of the component, since it doesn't depend on any props or state from inside.

Reply:
Makes sense. I have moved it out of the component in https://github.com/Automattic/wp-calypso/pull/74830/commits/c6282ae8da5186e657faa5a9458864dfd5d92b82.